Update to Discord V12.5.1 and synced changes with master branch

Fierythunder opened the pull request:
**Please describe the changes this PR makes and why it should be merged, as well as any and all proof of successful tests:**

All these changes are from master branch
- `README.md `
- `commands/help.js` 
- `commands/ping.js`
- `commands/set.js`
- `config.js.example`
- `events/guildCreate.js` added setActivity from master
- `events/guildDelete.js` setActivity from master
- `events/message.js`
- `events/guildMemberAdd.js`
- `index.js`
- `installer/setup.js`
- `util/functions.js`

My changes 
- `dashboard/dashboard.js` moved from `util/dashboard` so all dashboard files be in one place, updated few lines for discord v12
- `dashboard/templates/admin.ejs` fixed issues and made the list responsive
- `dashboard/templates/autherror.ejs` fixed footer link
- `dashboard/templates/blocks/footer.ejs` added the missing `</div>`
- `dashboard/templates/blocks/guild-card.ejs` removed fixed width and made few modifications
- `dashboard/templates/blocks/header.ejs` fixed avatar links nd moved navigation to separate file `dashboard/templates/blocks/navigation.ejs`
- `dashboard/templates/commands.ejs` fixed overlapping of long description
- `dashboard/templates/dashboard.ejs` made the list responsive and aligned
- `events/ready.js` updated to v12

[Live Example of Working Dashboard](https://guidebot.fierythunder.repl.co/)

**Semantic versioning classification:**  
- [ ] This PR changes the framework's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to README, etc.

AndrewJWin commented on LICENSE:
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2017 Évelyne Lachance
+Copyright (c) 2018 YorkAARGH
```
The license hasn't changed, this should remain as Évelyne.

Fierythunder replied:
Year 2018 is in one branch so kept the updated one

eslachance replied:
Cool. Don't change the **author** name, thanks.

Fierythunder replied:
Ok

AndrewJWin commented on README.md:
```diff
@@ -32,13 +46,13 @@ You also need to switch to the `dashboard` branch, which is done with the follow
 
 - In the folder from where you ran the git command, run `cd guidebot` and then run `npm install`
-- If any errors happen during install, check [Enmap's Docs](https://enmap.evie.codes/install) for troubleshooting tips.
+- **If you get any error about python or msibuild.exe or binding, read the requirements section again!**pop
```
Unnecessary "pop" at the end.

Fierythunder replied:
I will remove it,

AndrewJWin commented on commands/reboot.js:
```diff
@@ -1,9 +1,9 @@
-    await client.unloadCommand(cmd);
-  });
-  process.exit(1);
+  await Promise.all(client.commands.map(cmd =>
```
Please explain the difference between these lines.

Fierythunder replied:
I just changed it, thinking the master one is updated version

AndrewJWin commented on config.js.example:
```diff
@@ -1,96 +1,118 @@
+    { level: 10,
+      name: "Bot Owner", 
+      // Another simple check, compares the message author id to the one stored in the config file.
+      check: (message) => message.client.config.ownerID === message.author.id
```
Doesn't make use of the new Teams functionality, should automatically pull a list of "Owners" from the application.

Fierythunder replied:
team used in next commit

AndrewJWin commented on dashboard/app.js:
```diff
@@ -0,0 +1,26 @@
+  // This loop ensures that client.application always contains up to date data
+  // about the app's status. This includes whether the bot is public or not,
+  // its description, owner(s), etc. Used for the dashboard amongs other things.
+  client.application = await client.fetchApplication();
```
Why are we pulling the application here? This is for the dashboard, not the bot.

Fierythunder replied:
fixed this issue

AndrewJWin commented on events/ready.js:
```diff
@@ -1,21 +1,4 @@
 module.exports = async client => {
-  // Why await here? Because the ready event isn't actually ready, sometimes
```
Again, going back to where the application is going, rightfully it should be here so the settings can grab the application for the permission tree.

Fierythunder replied:
fixed it

AndrewJWin commented on installer/setup.js:
```diff
@@ -39,6 +39,31 @@ let prompts = [
     name: "ownerID",
     message: "Please enter the bot owner's User ID"
   },
+  {
```
Inventive use on allowing the user to select which intents to use in their bot.

AndrewJWin commented on util/functions.js:
```diff
@@ -56,14 +55,16 @@ module.exports = (client) => {
 
   // writeSettings overrides, or adds, any configuration item that is different
   // than the defaults. This ensures less storage wasted and to detect overrides.
+  const _ = require("lodash");
```
Why did you move the require to down here? By standard it should be on top. It was unnecessary to move this.

AndrewJWin reviewed (changes requested):
Overall, good work for adapting the code for V12. While the code is not actively maintained it is good to see activity here.

I would like to see some changes especially for gathering Owners from the Team in the Bot Application.

Included as well some documentation or explanation of some interesting changes.

AndrewJWin commented on index.js:
```diff
@@ -1,27 +1,32 @@
 // Node version, if it isn't it will throw the following error to inform
 // you.
-if (Number(process.version.slice(1).split(".")[0]) < 8) throw new Error("Node 8.0.0 or higher is required. Update Node on your system.");
+if (Number(process.version.slice(1).split(".")[0]) < 14) throw new Error("Node 14.0.0 or higher is required. Update Node on your system.");
```
Actually, while validating this, Discord.JS V12.5.1 only requires Node version 12.0.0 or higher. While yes, we should require users to use an LTS version - we shouldn't set it as the requirement.

Fierythunder replied:
Ok

Fierythunder commented:
> Overall, good work for adapting the code for V12. While the code is not actively maintained it is good to see activity here.
> 
> I would like to see some changes especially for gathering Owners from the Team in the Bot Application.
> 
> Included as well some documentation or explanation of some interesting changes.

Fierythunder commented:
I updated the changes, 
### About dashboard/app.js
1. Moved ```client.owners = new Array(); ``` from ```index.js```,
2. Moved ```client.application``` from ```events/ready.js```
Made these to changes so all files related to dashboard will be from this file instead of index.js and ready. 
Also moved ```client.wait``` from ```events/ready.js``` because this is needed for the client to fetch complete details from bot applications.

YorkAARGH commented on commands/set.js:
```diff
@@ -11,71 +11,70 @@
 exports.run = async (client, message, [action, key, ...value], level) => { // eslint-disable-line no-unused-vars
-  
 
-  // First we need to retrieve current guild settings
```
Not sure why you're changing the language used from warm and friendly to cold and clinical.

Fierythunder replied:
changed to previous

YorkAARGH commented on commands/set.js:
```diff
@@ -11,71 +11,70 @@
   const overrides = client.settings.get(message.guild.id);
   if (!client.settings.has(message.guild.id)) client.settings.set(message.guild.id, {});
   
-  // Secondly, if a user does `-set edit <key> <new value>`, let's change it
```
Same here.

YorkAARGH commented on commands/set.js:
```diff
@@ -11,71 +11,70 @@
     message.reply(`${key} successfully edited to ${joinedValue}`);
   } else
   
-  // If a user does `-set del <key>`, let's ask the user if they're sure...
```
Yet another IMHO pointless change to the language styling.

YorkAARGH commented on commands/set.js:
```diff
@@ -11,71 +11,70 @@
-
-      // We reset the `key` here.
+    if (["y", "yes"].includes(response.toLowerCase())) {
+      // We delete the `key` here.
```
The previous lines mention about "resetting" the key, but you change it to "delete", this could cause some confusion with newer developers.

YorkAARGH commented on commands/set.js:
```diff
@@ -11,71 +11,70 @@
     }
   } else
   
-  // Using `-set get <key>` we simply return the current value for the guild.
```
Why remove this?
You didn't even replace it.

YorkAARGH commented on dashboard/app.js:
```diff
@@ -0,0 +1,26 @@
+  // This loop ensures that client.application always contains up to date data
+  // about the app's status. This includes whether the bot is public or not,
+  // its description, owner(s), etc. Used for the dashboard amongs other things.
+  client.application = await client.fetchApplication();
```
This should remain in the ready event as this will end up being called before the Discord Bot/Application is ready and could result in some problems within the dashboard as you would need to wait for everything to populate.

YorkAARGH commented on dashboard/dashboard.js:
```diff
@@ -109,6 +109,10 @@ module.exports = (client) => {
   app.use(passport.initialize());
   app.use(passport.session());
+  
+  // Content Security Policy (CSP) is an added layer of security that helps to detect and mitigate certain types of attacks
```
Really not sure why this chunk of text was added...

Fierythunder replied:
reduced the text in next commit, this here because, in some browsers 3rd party usages and links are blocked because of helmet()

YorkAARGH commented on events/ready.js:
```diff
@@ -1,21 +1,4 @@
 module.exports = async client => {
-  // Why await here? Because the ready event isn't actually ready, sometimes
```
Legitimately you shouldn't be attempting to pull bot information if the bot isn't ready, undo this change.

Fierythunder replied:
changed back to previous in next commit

YorkAARGH commented on index.js:
```diff
@@ -1,27 +1,32 @@
 // client.config.prefix contains the message prefix
 
 // Require our logger
-client.logger = require("./util/logger.js");
```
Whilst I know this is perfectly valid, the change you made goes against pre-established notions (Line 12 for example), please revert this change.

Fierythunder replied:
ok

YorkAARGH commented on index.js:
```diff
@@ -32,14 +37,10 @@ require("./util/functions.js")(client);
 
-// Discord "recently" introduced Teams, where you can share bot applications between
-// a team of developers, so let's create a new array to push owner ids to.
-client.owners = new Array();
```
Why was this moved?

YorkAARGH commented on installer/setup_base.txt:
```diff
@@ -18,63 +18,82 @@ const config = {
   permLevels: [
-    {
-      level: 0,
+    // This is the lowest permisison level, this is for non-roled users.
```
You can still have a role and still be level 0, please remove the mention of `non-roled users`

Fierythunder replied:
ok

YorkAARGH commented on installer/setup_base.txt:
```diff
@@ -18,63 +18,82 @@ const config = {
       name: "Bot Owner", 
-      check: (message) => message.client.owners.includes(message.author.id)
+      // Another simple check, compares the message author id to the one stored in the config file.
+      check: (message) => message.client.config.ownerID === message.author.id
```
This does not take advantage of the teams that was implemented by Discord a little while ago. revert these changes.

YorkAARGH commented on package.json:
```diff
@@ -8,7 +8,7 @@
   },
   "engines": {
-    "node": ">=8"
+    "node": ">=14"
```
In the `index.js` file you told it to throw an error if the version wasn't v12, yet here you have told it to require v14 or greater, please fix for consistency.

Fierythunder replied:
ok

YorkAARGH reviewed (changes requested):
A lot of the feedback I have left is regarding the language used, we try to be warm, inviting and informal with how we teach, in more than one location you've managed to go from warm and inviting, to cold and clinical, that's not how we operate.

Also, a lot of changes you have proposed in your changes, make no sense **to me** and in your commit changes you don't mention __why__ you've made these changes.

ItsMrReals commented:
Pretty good, there is some html bugs like these
![firefox_ljcGOv11Nr](https://user-images.githubusercontent.com/60827052/108003496-3a6bc280-6fb0-11eb-94be-89141cb6099e.png)
![firefox_BC46Dc2tTN](https://user-images.githubusercontent.com/60827052/108003498-3b045900-6fb0-11eb-8394-cd4354629d0a.png)

YorkAARGH commented:
@ItsMrReals you should create your own issue thread instead of attempting to hijack this PR for support.

ItsMrReals commented:
> @ItsMrReals you should create your own issue thread instead of attempting to hijack this PR for support.

I was not asking for support, this is an issue with his branch, not the main one, I was suggesting some fixes

YorkAARGH commented:
It honestly came across as a request support, as you did not mention in your initial post that it was a suggestion for fixes.

eslachance commented:
@Fierythunder Do you intend to update as was requested by York? If not I'll need to close this PR.

Fierythunder commented:
i will update the changes within 2-3 days

Fierythunder reviewed (commented):
i updated and fixed and changed back

Fierythunder commented:
@YorkAARGH @eslachance i made the requested changes! Please review again!

YorkAARGH commented on README.md:
```diff
@@ -49,10 +50,13 @@ You also need to switch to the `dashboard` branch, which is done with the follow
-- **If you get any error about python or msibuild.exe or binding, read the requirements section again!**
+- If any errors happen during install, check [Enmap's Docs](https://enmap.evie.codes/install) for troubleshooting tips.
 - Once the installation is done, the setup will automatically run. Follow the instructions given to finalize everything.
+- **If you get any error about python or msibuild.exe or binding, read the requirements section again!**
```
I would say combine this, and line 53, as you can get errors from other packages and would **possibly** confuse newer developers if they can't find their issue on Evie's site.

YorkAARGH commented on commands/conf.js:
```diff
@@ -76,12 +78,7 @@ exports.run = async (client, message, [action, key, ...value], level) => { // es
-    });
-    await message.channel.send(`= Bot Default Settings =
-${array.join("\n")}`, { code: "asciidoc" });
+    await message.channel.send(`***__Bot Default Settings__***\n\`\`\`json\n${inspect(defaults)}\n\`\`\``);
```
Whilst absolutely valid, it would destroy the formatting of the help command. We've padded/aligned things to be more easily readable and I'm not sure why you're changing it from asciidoc to json as it doesn't look as aesthetically pleasing when posted, please revert this change.

Fierythunder replied:
I made this change from "master" branch, 
Ok i will revert it back

YorkAARGH replied:
Hmm, on closer inspection you do seem to be correct, not sure why (in the master branch) it was changed.

Thank you for bringing this to my attention.

YorkAARGH commented on commands/help.js:
```diff
@@ -18,17 +21,17 @@ exports.run = (client, message, args, level) => {
-      output += `${message.settings.prefix}${c.help.name}${" ".repeat(longest - c.help.name.length)} :: ${c.help.description}\n`;
+      output += `${settings.prefix}${c.help.name}${" ".repeat(longest - c.help.name.length)} :: ${c.help.description}\n`;
     });
-    message.channel.send(output, {code: "asciidoc", split: { char: "\u200b" }});
```
This split shouldn't have been removed at all as a bot could have a large number of commands, too many to list in a single message, my old bot Misaki for example had over 200 commands and this singular change would cause the help command to error. Please revert this change.

Fierythunder replied:
Ok

YorkAARGH commented on commands/set.js:
```diff
@@ -11,70 +11,74 @@
     if (!defaults[key]) return message.reply("This key does not exist in the settings");
     const isDefault = !overrides[key] ? "\nThis is the default global default value." : "";
-    message.reply(`The value of ${key} is currently ${settings[key]}${isDefault}`);
+    message.replymessage.reply(`The value of ${key} is currently ${settings[key]}${isDefault}`);
```
This needs fixing, you did a goof.

Fierythunder replied:
Oops

YorkAARGH commented on dashboard/templates/admin.ejs:
```diff
@@ -1,44 +1,32 @@
 <%- include('blocks/header', {bot, user, path}) %>
```
I do not know what you have done to this page, as I cannot access it **at all** when I launch the bot with these changes. Including going directly to the URL.

Fierythunder replied:
I will add a live dashboard link

Fierythunder replied:
found the issue

YorkAARGH commented on dashboard/templates/blocks/header.ejs:
```diff
@@ -31,52 +31,7 @@
 
 <body>
   <a name="top"></a>
-  <nav class="navbar navbar-expand-lg navbar-dark bg-primary">
```
Not sure why you felt the need to remove all of this code, as it completely ruins the look and design.

[Currently available](https://i.imgur.com/WLJyGo3.png)
[Your proposed changes](https://i.imgur.com/uIv4YIu.png)

Fierythunder replied:
All because of the container, i will change it

YorkAARGH commented on dashboard/templates/commands.ejs:
```diff
@@ -16,7 +16,7 @@ bot.commands.forEach((command) => {
       <% for (const command of help[category]) { %>
         <div class="card">
           <h3 class="card-header"><%= command.help.name %></h3>
-          <div class="card-body" style="height: 100px;">
```
Revert this change, it makes the command page look unorganized and cluttered.
[Current](https://i.imgur.com/vAYObto.png)
[Your proposed changes](https://i.imgur.com/qKYbI3q.png)

Fierythunder replied:
If description is long then it overlaps the footer, so removed the height

YorkAARGH replied:
Whilst true, it gave a uniformed appearance.

Maybe look into limiting characters/words so they fit without causing issues.

Fierythunder replied:
max 60 characters allowed

YorkAARGH commented on dashboard/templates/dashboard.ejs:
```diff
@@ -1,79 +1,53 @@
-            <h6><%= guild.name %></h6>
-            <a href="<%= href %>"> <%= text %> <i class="fa fa-fw fa-chevron-circle-right" aria-hidden="true"></i></a>
-          </div>
+<h1 class="display-3">Dashboard</h1>
```
I can see what you've done, and I do like it, however please align the servers, otherwise it will look cluttered and unorganised.

Also, the MANAGE link does not work, it keeps taking me to the home page.

YorkAARGH replied:
This is the error that is thrown when I try to access the server manage page.

```
TypeError: C:\Users\YorkAARGH\Desktop\Dashboards\New\dashboard\templates\guild\manage.ejs:8
    6|   <div class="row" style="margin-top: 25px;">
    7|     <div class="col">
 >> 8|       <%- include('../blocks/guild-card', {guild}) %>
    9|     </div>
    10|     <div class="col">
    11|       <form method="post">

C:\Users\YorkAARGH\Desktop\Dashboards\New\dashboard\templates\blocks\guild-card.ejs:24
    22|       </li>
    23|       <li class="list-group-item d-flex justify-content-between align-items-center" style="width: 100%" data-toggle="tooltip" title="Server Owner.">
 >> 24|         <i class="fa fa-fw fa-user-circle-o" aria-hidden="true"></i>Owner: <span id="owner"><%= `${guild.owner.user.tag}` %></span>
    25|       </li>
    26|         <li class="list-group-item d-flex justify-content-between align-items-center" data-toggle="tooltip" title="Members">
    27|           <i class="fa fa-fw fa-users" aria-hidden="true"></i>Members <span class="badge badge-primary badge-pill"><%= `${guild.memberCount}` %></span>

Cannot read property 'user' of null
    at eval (eval at compile (C:\Users\YorkAARGH\Desktop\Dashboards\New\node_modules\ejs\lib\ejs.js:662:12), <anonymous>:40:41)
    at guild-card (C:\Users\YorkAARGH\Desktop\Dashboards\New\node_modules\ejs\lib\ejs.js:692:17)
    at include (C:\Users\YorkAARGH\Desktop\Dashboards\New\node_modules\ejs\lib\ejs.js:690:39)
    at eval (eval at compile (C:\Users\YorkAARGH\Desktop\Dashboards\New\node_modules\ejs\lib\ejs.js:662:12), <anonymous>:14:17)
    at manage (C:\Users\YorkAARGH\Desktop\Dashboards\New\node_modules\ejs\lib\ejs.js:692:17)
    at tryHandleCache (C:\Users\YorkAARGH\Desktop\Dashboards\New\node_modules\ejs\lib\ejs.js:272:36)
    at View.exports.renderFile [as engine] (C:\Users\YorkAARGH\Desktop\Dashboards\New\node_modules\ejs\lib\ejs.js:489:10)
    at View.render (C:\Users\YorkAARGH\Desktop\Dashboards\New\node_modules\express\lib\view.js:135:8)
    at tryRender (C:\Users\YorkAARGH\Desktop\Dashboards\New\node_modules\express\lib\application.js:640:10)
    at Function.render (C:\Users\YorkAARGH\Desktop\Dashboards\New\node_modules\express\lib\application.js:592:3)
TypeError: C:\Users\YorkAARGH\Desktop\Dashboards\New\dashboard\templates\guild\manage.ejs:8
    6|   <div class="row" style="margin-top: 25px;">
    7|     <div class="col">
 >> 8|       <%- include('../blocks/guild-card', {guild}) %>
    9|     </div>
    10|     <div class="col">
    11|       <form method="post">
```

Fierythunder replied:
Ok, i will check

Fierythunder replied:
its because of `guild.owner.user.tag` and it can be fixed with Intents for cache

YorkAARGH commented on dashboard/templates/guild/members.ejs:
```diff
@@ -81,8 +81,8 @@
       member = members[i];
-      var color = member.roles.highest.hexColor === "#000000" ? '#FFFFFF' : member.roles.highest.hexColor;
-      var roles = member.roles.cache.filter(r=>r.name !== "@everyone").map(r=> {
+      var color = member.highestRole.hexColor === "#000000" ? '#FFFFFF' : member.highestRole.hexColor;
```
Revert this change as `highestRole` no longer exists.

Fierythunder replied:
It has no issues, [check this line](https://github.com/Fierythunder/guidebot/blob/878e2f04e65d5fea8c63711f73b9f268607a3f13/dashboard/dashboard.js#L329)

YorkAARGH replied:
Oh, I completely forgot that was a thing, I checked against the discord.js documentation and `highestRole` was no longer available. My apologies.

Fierythunder replied:
in previous commit i also didn't see this line
so the members list didn't show up

YorkAARGH commented on events/guildCreate.js:
```diff
@@ -2,7 +2,7 @@
 
 module.exports = (client, guild) => {
   // Update the activity to show the updated number of guilds.
-  client.user.setActivity(`${client.settings.get("default").prefix}help | ${client.guilds.cache.size} Servers`);
```
Not sure why this was changed.

Fierythunder replied:
As we already have getSettings in function.js

YorkAARGH commented on events/guildCreate.js:
```diff
@@ -2,7 +2,7 @@
 module.exports = (client, guild) => {
   // Update the activity to show the updated number of guilds.
-  client.user.setActivity(`${client.settings.get("default").prefix}help | ${client.guilds.cache.size} Servers`);
+  client.user.setActivity(`${client.getSettings("default").prefix}help | ${client.guilds.cache.size} Servers`, {type: "PLAYING"});
```
Not sure why you had to add the playing option as status messages default to playing.

Fierythunder replied:
under default,  it was none in [Discord.js/setActivity](https://discord.js.org/#/docs/main/stable/class/ClientUser?scrollTo=setActivity)

YorkAARGH replied:
Yet I tested the original code that does not have `PLAYING` and it defaulted to `PLAYING`, which is why I mentioned it's a needless change.

Unless it changed in 12.5.

[VSCode search](https://i.imgur.com/07jE2XE.png)
[Bot status](https://i.imgur.com/zLwbe5N.png)

Fierythunder replied:
i just checked the docs and the master branch

YorkAARGH replied:
I stand corrected, we did in fact add it to the master branch, disregard what I said.

YorkAARGH commented on events/guildDelete.js:
```diff
@@ -1,14 +1,15 @@
+  if (!guild.available) return; // If there is an outage, return.
   // Update the activity to show the updated number of guilds.
-  client.user.setActivity(`${client.settings.get("default").prefix}help | ${client.guilds.cache.size} Servers`);
+  client.user.setActivity(`${client.getSettings("default").prefix}help | ${client.guilds.cache.size} Servers`, {type: "PLAYING"});
```
Same here, not sure why you had to add the playing type as that is the default.

Fierythunder replied:
under default,  it was none in [Discord.js/setActivity](https://discord.js.org/#/docs/main/stable/class/ClientUser?scrollTo=setActivity)

YorkAARGH commented on events/message.js:
```diff
@@ -16,10 +17,9 @@ module.exports = async (client, message) => {
   // Just in case we don't know what the current prefix is, mention the bot
   // and the following regex will detect it and fire off letting you know
   // what the current prefix is.
-  // Checks if the bot was mentioned, with no message after it, returns the prefix.
```
This honestly seems to be a pointless change.
Please revert.

YorkAARGH commented on events/message.js:
```diff
@@ -67,6 +68,23 @@ module.exports = async (client, message) => {
     }
   }
 
+  //Cooldowns 
```
Why was this added?
It's not featured in any production code.

Fierythunder replied:
This example is from discord.js guide

YorkAARGH replied:
That may be, but the aim of this PR was to update and fix issues is it not?

Fierythunder replied:
only one addition
should i remove it

YorkAARGH replied:
To be honest, yes. As this feature isn't in any of the other branches/repositories.

Fierythunder replied:
ok

YorkAARGH reviewed (changes requested):
I'm not going to lie, a lot of these changes appear pointless, or stuff you have added to your bot and wanting to PR it into this.

It also comes across that you haven't tested half of your proposed changes, the admin page is removed/combined with the manage server page, which doesn't work, the dashboard page whilst does look a lot better than what it did looks cluttered and unorganised, and same with the commands page.

You've also some how broken the site header as everything is supposed to be in a singular column in the middle of the browser.

Please make the suggested changes.

Also, please actually describe what your PR is doing, "Fixed some Dashboard webpage issues" is rather vague and on closure inspection i.e. testing, it looks like you've introduced needless changes and more issues.

Fierythunder commented:
All these changes are from master branch
- `README.md `
- `commands/help.js` 
- `commands/ping.js`
- `commands/set.js`
- `config.js.example`
- `events/guildCreate.js` added setActivity from master
- `events/guildDelete.js` setActivity from master
- `events/message.js`
- `events/guildMemberAdd.js`
- `index.js`
- `installer/setup.js`
- `util/functions.js`

My changes 
- `dashboard/dashboard.js` moved from `util/dashboard` so all dashboard files be in one place, updated few lines for discord v12
- `dashboard/templates/admin.ejs` fixed issues and made the list responsive
- `dashboard/templates/autherror.ejs` fixed footer link
- `dashboard/templates/blocks/footer.ejs` added the missing `</div>`
- `dashboard/templates/blocks/guild-card.ejs` removed fixed width and made few modifications
- `dashboard/templates/blocks/header.ejs` fixed avatar links nd moved navigation to separate file `dashboard/templates/blocks/navigation.ejs`
- `dashboard/templates/commands.ejs` fixed overlapping of long description
- `dashboard/templates/dashboard.ejs` made the list responsive and aligned
- `events/ready.js` updated to v12

[Live Example of Working Dashboard](https://guidebot.fierythunder.repl.co/)
@YorkAARGH

Fierythunder commented:
Any Corrections? @YorkAARGH @eslachance

Codeize commented:
Is this still active or will it be ignored?

Fierythunder commented:
Waiting for the review

YorkAARGH commented:
Hi sorry, I completely forgot about this, I'll be reviewing this in a couple of days, a lot of stuff has happened in my life atm.

YorkAARGH commented:
ah nuts... I completely forgot about this *AGAIN*... honestly, I don't think there'd be much point in reviewing/merging this right now as it will require **another** update in like a week or so when version 13 drops.
